source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -46,6 +47,23 @@ DnsCacheImpl::~DnsCacheImpl() {
   }
 }
 
+Network::DnsResolverSharedPtr DnsCacheImpl::selectDnsResolver(
+    const envoy::extensions::common::dynamic_forward_proxy::v3::DnsCacheConfig& config,
+    Event::Dispatcher& main_thread_dispatcher) {
+  if (config.has_dns_resolver() && !config.dns_resolver().resolvers().empty()) {
+    const auto& resolver_addrs = config.dns_resolver().resolvers();
+    std::vector<Network::Address::InstanceConstSharedPtr> resolvers;
+    resolvers.reserve(resolver_addrs.size());
+    for (const auto& resolver_addr : resolver_addrs) {
+      resolvers.push_back(Network::Address::resolveProtoAddress(resolver_addr));
+    }
+    const bool use_tcp_for_dns_lookups = config.use_tcp_for_dns_lookups();
```

Comment:
nit: can pass directly to the parameter.

Reply:
Hi @adisuissa, since the PR #16237 will follow this PR(16294) shortly, this line will be replaced by using `dns_resolver_options`. Can this PR get merged as it is?